src/test/java/org/jenkinsci/plugins/workflow/support/steps/StageStepTest.java

```diff
@@ -67,6 +67,19 @@
         });
     }
 
+    @Issue("JENKINS-44456")
+    @Test public void stageNameInEnv() throws Exception {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition("stage('foo-stage') {echo \"STAGE_NAME is ${STAGE_NAME}\"}", true));
+                WorkflowRun b = story.j.assertBuildStatusSuccess(p.scheduleBuild2(0));
```

Comment:
Could add another `step` to prove that the serialization works, but probably overkill here.